query-connector/src/app/queryBuilding/components/ConditionSelection.tsx

```diff
@@ -0,0 +1,201 @@
+    SetStateAction<CategoryNameToConditionOptionMap | undefined>
+  >;
+  queryName: string;
+  validateForm: () => void;
+  setFormError: Dispatch<SetStateAction<FormError>>;
+  formError: FormError;
+  setLoading: Dispatch<SetStateAction<boolean>>;
+  loading: boolean;
+  setConditionValueSets: Dispatch<
+    SetStateAction<ConditionIdToValueSetArray | undefined>
+  >;
+};
+
+/**
+ * Display component for a condition on the query building page
+ * @param root0 - params
+ * @param root0.fetchedConditions - ID of the condition to reference
+ * @param root0.selectedConditions - name of condition to display
+ * @param root0.setBuildStep - Redirect function to handle view routing
+ * @param root0.setFetchedConditions - listener function for checkbox
+ * selection
+ * @param root0.setSelectedConditions - current checkbox selection status
+ * @param root0.queryName - current checkbox selection status
+ * @param root0.validateForm - function that checks for a valid query name
+ * and at least one selected condition
+ * @param root0.formError - indicates missing or incorrect form data
+ * @param root0.setFormError - state function that updates the status of the
+ * condition selection form input data
+ * @param root0.loading -  Boolean to track loading state
+ * @param root0.setLoading - The function to set the loading state
+ * @param root0.setConditionValueSets - state function that updates the status of the
+ * condition selection form input data
+ * @returns A component for display to redner on the query building page
+ */
+export const ConditionSelection: React.FC<ConditionSelectionProps> = ({
+  fetchedConditions,
+  selectedConditions,
+  setBuildStep,
+  setFetchedConditions,
+  setSelectedConditions,
+  queryName,
+  validateForm,
+  formError,
+  setFormError,
+  loading,
+  setConditionValueSets,
+  setLoading,
+}) => {
+  const focusRef = useRef<HTMLInputElement | null>(null);
+
+  const [searchFilter, setSearchFilter] = useState<string>();
+
+  useEffect(() => {
+    let isSubscribed = true;
+
+    if (queryName == "" || queryName == undefined) {
+      focusRef?.current?.focus();
+    }
+
+    async function fetchConditionsAndUpdateState() {
+      const { categoryToConditionArrayMap } = await getConditionsData();
+
+      if (isSubscribed) {
+        setFetchedConditions(
+          mapFetchedDataToFrontendStructure(categoryToConditionArrayMap),
+        );
+      }
+    }
+
+    fetchConditionsAndUpdateState().catch(console.error);
+    return () => {
+      isSubscribed = false;
+    };
+  }, []);
+
+  async function getValueSetsForSelectedConditions() {
+    const conditionIds = Object.entries(selectedConditions)
+      .map(([_, conditionObj]) => {
+        return Object.keys(conditionObj);
+      })
+      .flatMap((ids) => ids);
+
+    const ConditionValueSets: ConditionIdToValueSetArray = {};
+
+    for (const id of conditionIds) {
+      const results: ValueSet[] =
+        await getValueSetsAndConceptsByConditionID(id);
```

Comment:
we might want to construct all the promises and then batch resolve them using `Promise.allSettled` instead of iterating through and awaiting for each condition promise to resolve like we're doing in [the database generation scripts.](https://github.com/CDCgov/dibbs-query-connector/blob/7844330bce99e146cf9404c460ad7beff9a21b97/query-connector/src/app/database-service.ts#L312)

Reply:
per our convo today, this has been updated to pass the entire array of IDs to the db query in one go, so we avoid unnecessary db calls